### PR TITLE
Metrics transform processor: replace version in label values

### PR DESCRIPTION
**Pull request by james-bebbington**
Replace `{{version}}` in any label values in the metrics transform processor with the app version.

### CODE REVIEW

File: processor/metricstransformprocessor/README.md [hunk 1]
```diff
@@ -103,34 +104,31 @@ operations:
...
-      action: update
-      operations:
-        - action: add_label
-          new_label: mylabel
-          new_value: myvalue
+# add label `version` with value `opentelemetry collector v0.0.0` to all points
+operations:
+  - action: add_label
+    new_label: version
+    new_value: opentelemetry collector {{version}}
```
Comment (nilebox) [inline processor/metricstransformprocessor/README.md:131]: nit: Ideally there should be a way to escape this and be able to pass the `{{version}}` string as is (without replacement).
Though in practice I doubt anyone would ever run into this limitation.

Comment (nilebox) [reply processor/metricstransformprocessor/README.md:131]: I guess it's possible that some other processor also does substitution for `{{version}}`, but in a different way / with a different value. In that case the ordering of processors will matter.
Still don't think this is a real issue though.

File: processor/metricstransformprocessor/README.md [hunk 2]
```diff
@@ -103,34 +104,31 @@ operations:
...
-transforms:
 ...
-# The following will append label {Key: `mylabel`, Description: `myvalue`} to the metric `some_name`.
-  - metric_name: some_name
-      action: update
-      operations:
-        - action: add_label
-          new_label: mylabel
-          new_value: myvalue
+# add label `version` with value `opentelemetry collector v0.0.0` to all points
```
Comment (nilebox) [inline processor/metricstransformprocessor/README.md:127]:
```suggestion
# add label `version` with value `opentelemetry collector vX.Y.Z` to all points
```

**Review by nilebox — APPROVED** [review]

File: processor/metricstransformprocessor/README.md (hunk 1 above)
Comment (james-bebbington) [reply processor/metricstransformprocessor/README.md:131]: Yea this is a very simple substitution so I only defined trivial semantics. Note it only applies the configured values themselves. I may come back and add something more sophisticated later.

**Pull request merged**